### PR TITLE
Enhance Filtering and fix CI

### PR DESCRIPTION
**Pull request by codesmith14**
* Enhanced metric filtering to include machine and container spec metrics
* Fix Travis CI
* Add tests for Circle CI

### CODE REVIEW

File: converter/sfxConverter.go [hunk 1]
```diff
@@ -452,16 +469,170 @@ func NewCadvisorCollector(infoProvider infoProvider, f ContainerNameToLabelsFunc
...
+			},
+			getValues: func(machineInfo *info.MachineInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(machineInfo.MemoryCapacity))}}
+			},
+		},
+	}
+	var index = 0
+	for _, metric := range metrics {
+		// check if metric is not on the exclusion list
+		if _, ok := excludedMetrics[metric.name]; !ok {
```
Comment (jrcamp) [inline converter/sfxConverter.go:584]: Trying to figure out how this works. If metrics[0] is not excluded then we check metrics[1] and it is excluded is index not still equal to 0 and will overwrite metrics[0]?

File: converter/sfxConverter.go [hunk 2]
```diff
@@ -452,16 +469,170 @@ func NewCadvisorCollector(infoProvider infoProvider, f ContainerNameToLabelsFunc
...
+		}
+	}
+	// trim metrics down to the desired length
+	metrics = metrics[:index]
+	return metrics
+}
+
+func getContainerSpecMetrics(excludedMetrics map[string]bool) []containerSpecMetric {
+	var metrics = []containerSpecMetric{
+		// ch <- *datapoint.New("container_start_time_seconds", copyDims(dims), datapoint.NewIntValue(container.Spec.CreationTime.Unix()), datapoint.Gauge, tt)
```
Comment (jrcamp) [inline converter/sfxConverter.go]: Can this be removed?

File: converter/sfxConverter.go (hunk 1 above)
Comment (codesmith14) [reply converter/sfxConverter.go:584]: If the metric is excluded we we don't reassign it to the current index and  we don't increment index.  Then if the next metric is included we set the current index to that metric and increment the index.  This way we put all "included" metrics in the front of the metrics array 0...index.  Then we slice the array on index to get rid of all of the "excluded" metrics.

File: converter/sfxConverter.go (hunk 2 above)
Comment (codesmith14) [reply converter/sfxConverter.go]: yes it can, will do

**Review by jrcamp — APPROVED** [review]

**Pull request merged**